### PR TITLE
Sqlite: Implement metadata

### PR DESCRIPTION
**Pull request by bricelam**
Resolves: #875

Note, this uses the new syntax we discussed in API Review today. Fore example:
```C#
modelBuilder.Entity<Person>().SqliteTable("people");
```

Also note, `entityType.Sqlite().Schema` and `model.Sqlite().Sequences` return `null` and empty since those concepts don't exist on SQLite.

### CODE REVIEW

File: src/EntityFramework.Sqlite/Metadata/SqliteMetadataExtensionProvider.cs [hunk 1]
```diff
@@ -8,12 +8,11 @@ namespace Microsoft.Data.Entity.Sqlite.Metadata
...
     public class SqliteMetadataExtensionProvider : IRelationalMetadataExtensionProvider
     {
-        // TODO: Update with #875
-        public virtual IRelationalEntityTypeExtensions Extensions(IEntityType entityType) => entityType.Relational();
-        public virtual IRelationalForeignKeyExtensions Extensions(IForeignKey foreignKey) => foreignKey.Relational();
-        public virtual IRelationalIndexExtensions Extensions(IIndex index) => index.Relational();
-        public virtual IRelationalKeyExtensions Extensions(IKey key) => key.Relational();
-        public virtual IRelationalModelExtensions Extensions(IModel model) => model.Relational();
-        public virtual IRelationalPropertyExtensions Extensions(IProperty property) => property.Relational();
+        public virtual IRelationalEntityTypeExtensions Extensions(IEntityType entityType) => entityType.Sqlite();
```
Comment (lajones) [inline src/EntityFramework.Sqlite/Metadata/SqliteMetadataExtensionProvider.cs]: Should these return the ISqlliteXXXExtensions instead?

**Comment by lajones** [comment]
:shipit:

File: src/EntityFramework.Sqlite/Metadata/SqliteMetadataExtensionProvider.cs (hunk 1 above)
Comment (bricelam) [inline src/EntityFramework.Sqlite/Metadata/SqliteMetadataExtensionProvider.cs]: No; they're defined on the interface which is a Relational concept.

**Comment by ErikEJ** [comment]
@bricelam I am trying to understand the purpose of implementing all this in each provider, can you "enlighten" me ? :-)

**Comment by bricelam** [comment]
It's so that you can override things on a per-provider basis. E.g. one table name on SQLite, one on SQL Server, and one everywhere else. @rowanmiller commented on Friday that we may want to consider just removing it.

**Comment by ErikEJ** [comment]
@bricelam  Thanks for the heads up, I will just leave it for now

**Pull request merged**